vite.config.js

```diff
@@ -0,0 +1,159 @@
+import { defineConfig, loadEnv, transformWithEsbuild } from 'vite'
+import react from '@vitejs/plugin-react'
+import { createHtmlPlugin } from 'vite-plugin-html'
+import { sentryVitePlugin } from '@sentry/vite-plugin'
+import { whiteListEnv } from './whiteListEnv'
+import { execSync } from 'child_process'
+
+const defaultExtensions = ['.mjs', '.js', '.mts', '.ts', '.jsx', '.tsx', '.json']
+const allExtensions = [
+  ...defaultExtensions.map((ext) => ext.replace(/^\./, '.web.')),
+  ...defaultExtensions,
+]
+
+const libsThatHaveJSFilesContainingJSX = [
+  'node_modules/react-native-animatable',
+  'node_modules/react-native-qrcode-svg',
+  'node_modules/@ptomasroos/react-native-multi-slider',
+  'node_modules/react-native-calendars',
+  'node_modules/react-native-swipe-gestures',
+]
+
+const packageJson = require('./package.json')
+
+function getGitInfo(command) {
+  try {
+    return execSync(command).toString().trim()
+  } catch (e) {
+    console.error(`Failed to execute command: ${command}`, e)
+    return 'unknown'
+  }
+}
+
+const branch = getGitInfo('git rev-parse --abbrev-ref HEAD')
+const commitHash = getGitInfo('git rev-parse --short HEAD')
+
+export default ({ mode }) => {
+  const isDevMode = mode === 'development'
+  const isProdMode = mode === 'production'
+  const env = loadEnv(isDevMode ? 'testing' : mode, process.cwd(), '')
+  const proxyConfig = {
+    proxy: {
+      '/native': {
+        target: env.API_BASE_URL,
+        changeOrigin: true,
+      },
+      '/saml': {
+        target: env.API_BASE_URL,
+        changeOrigin: true,
+      },
+    },
+  }
+  return defineConfig({
+    define: {
+      global: 'window',
+      'process.env': whiteListEnv(env), // Do not expose the global object directly
+      'process.env.COMMIT_HASH': JSON.stringify(commitHash),
+      'process.env.BRANCH': JSON.stringify(branch),
+      __DEV__: env.NODE_ENV !== 'production',
+    },
+    plugins: [
+      react(),
+      {
+        apply: 'build', // This plugin runs only when building (not serve)
+        name: 'treat-js-files-as-jsx',
+        async transform(code, id) {
+          if (!libsThatHaveJSFilesContainingJSX.some((lib) => id.includes(lib))) return null
```

Comment:
dans ce cas, ça revient au meme, c'est du détail

```suggestion
          if (!libsThatHaveJSFilesContainingJSX.some((lib) => id.endsWith(lib))) return null
```